src/lib/stable/ic_apis/call.ts

```diff
@@ -32,77 +74,142 @@ import { idlDecode, idlEncode } from '../execute_with_candid_serde';
  *   - after a successful inter-canister await from a composite query
  *   - after an unsuccessful inter-canister await from a composite query
  */
-export async function call<Args extends any[] | undefined, Return = any>(
+export async function call<
+    Args extends any[] | Uint8Array | undefined,
+    Return = any
+>(
     canisterId: Principal | string,
     method: string,
-    options?: {
-        paramIdlTypes?: IDL.Type[];
-        returnIdlType?: IDL.Type;
-        args?: Args;
-        cycles?: bigint;
-        raw?: Uint8Array;
-    }
+    options?: CallOptions<Args>
 ): Promise<Return> {
-    // TODO this should use a Result remember
-    return new Promise((resolve, reject) => {
-        if (
-            globalThis._azleIcStable === undefined &&
-            globalThis._azleIcExperimental === undefined
-        ) {
-            return undefined;
+    if (typeof options?.timeout === 'bigint') {
+        throw new Error('timeout is not yet implemented');
+    }
+
+    const canisterIdBytes = getCanisterIdBytes(canisterId);
+    const argsRaw = getArgsRaw(options);
+    const cyclesString = getCyclesString(options);
+
+    if (options?.oneway === true) {
+        return handleOneWay<Return>(
+            canisterIdBytes,
+            method,
+            argsRaw,
+            cyclesString
+        );
+    } else {
+        return handleTwoWay<Return>(
+            canisterIdBytes,
+            method,
+            argsRaw,
+            cyclesString,
+            options?.raw ?? false,
+            options?.returnIdlType
+        );
+    }
+}
+
+function getCanisterIdBytes(canisterId: Principal | string): Uint8Array {
+    return typeof canisterId === 'string'
+        ? Principal.fromText(canisterId).toUint8Array()
+        : canisterId.toUint8Array();
+}
+
+function getArgsRaw<Args extends any[] | Uint8Array | undefined>(
+    callOptions?: CallOptions<Args>
+): Uint8Array {
+    if (callOptions?.raw === true) {
+        if (callOptions?.args === undefined) {
+            return idlEncode([], []);
+        }
+
+        if (callOptions.args instanceof Uint8Array === false) {
+            throw new Error(
+                `args must be a Uint8Array. If you did not intend to make a raw call, then consider setting the raw property of the call options to undefined or false`
+            );
+        }
+
+        return callOptions.args;
+    } else {
+        if (callOptions?.args instanceof Uint8Array === true) {
+            throw new Error(
+                `args must be an array of JavaScript values. If you intended to make a raw call, then consider setting the raw property of the call options to true`
+            );
         }
 
+        return idlEncode(
```

Comment:
I guess we could also check if both paramIdlTypes and args are undefined then just return the preencoded (), but that seems like it could be a bit of overkill in this case, but I thought I would throw out the idea and see what you think

Reply:
Done